libarchive/archive_read_support_filter_lz4.c

```diff
@@ -344,31 +344,33 @@ lz4_filter_read(struct archive_read_filter *self, const void **p)
 			state->eof = 1;
 			*p = NULL;
 			return (0);
-		}
-		uint32_t number = archive_le32dec(read_buf);
```

Comment:
Here and through the rest of the patch - do not re-indent hunks of code, just move the declaration further up.

You'd want to enable `-Wdeclaration-after-statement` to catch these issues.